src/config.rs

```diff
@@ -47,42 +46,40 @@ pub struct Configuration<E = LittleEndian, I = Varint, A = SkipFixedArrayLength,
     _l: PhantomData<L>,
 }
 
-impl Configuration {
-    /// The default config for bincode 2.0. By default this will be:
-    /// - Little endian
-    /// - Variable int encoding
-    /// - Skip fixed array length
-    pub const fn standard() -> Self {
-        Self::generate()
-    }
+/// The default config for bincode 2.0. By default this will be:
+/// - Little endian
+/// - Variable int encoding
+/// - Skip fixed array length
+pub const fn standard() -> Configuration {
+    generate()
+}
 
-    /// Creates the "legacy" default config. This is the default config that was present in bincode 1.0
-    /// - Little endian
-    /// - Fixed int length encoding
-    /// - Write array lengths
-    pub const fn legacy() -> Configuration<LittleEndian, Fixint, WriteFixedArrayLength, NoLimit> {
-        Self::generate()
-    }
+/// Creates the "legacy" default config. This is the default config that was present in bincode 1.0
+/// - Little endian
+/// - Fixed int length encoding
+/// - Write array lengths
+pub const fn legacy() -> Configuration<LittleEndian, Fixint, WriteFixedArrayLength, NoLimit> {
+    generate()
 }
 
-impl<E, I, A, L> Configuration<E, I, A, L> {
-    const fn generate<_E, _I, _A, _L>() -> Configuration<_E, _I, _A, _L> {
-        Configuration {
-            _e: PhantomData,
-            _i: PhantomData,
-            _a: PhantomData,
-            _l: PhantomData,
-        }
+const fn generate<_E, _I, _A, _L>() -> Configuration<_E, _I, _A, _L> {
+    Configuration {
+        _e: PhantomData,
+        _i: PhantomData,
+        _a: PhantomData,
+        _l: PhantomData,
     }
+}
 
+impl<E, I, A, L> Configuration<E, I, A, L> {
     /// Makes bincode encode all integer types in big endian.
     pub const fn with_big_endian(self) -> Configuration<BigEndian, I, A> {
```

Comment:
Unrelated to this PR in particular, but I just noticed that this doesn't specify `L` in the return type. Does this mean we are erasing the limits?

Reply:
Yes...

fixed and added a comment on adding more options